src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Tests/Functions/DeleteDataSetVersionFunctionTests.cs

```diff
@@ -0,0 +1,433 @@
+
+public abstract class DeleteDataSetVersionFunctionTests(ProcessorFunctionsIntegrationTestFixture fixture)
+    : ProcessorFunctionsIntegrationTest(fixture)
+{
+    public class DeleteDataSetVersionTests : DeleteDataSetVersionFunctionTests
+    {
+        private readonly IDataSetVersionPathResolver _dataSetVersionPathResolver;
+
+        public DeleteDataSetVersionTests(ProcessorFunctionsIntegrationTestFixture fixture) : base(fixture)
+        {
+            _dataSetVersionPathResolver = GetRequiredService<IDataSetVersionPathResolver>();
+        }
+
+        [Theory]
+        [InlineData(DataSetVersionStatus.Failed)]
+        [InlineData(DataSetVersionStatus.Mapping)]
+        [InlineData(DataSetVersionStatus.Draft)]
+        [InlineData(DataSetVersionStatus.Cancelled)]
+        public async Task Success_FirstDataSetVersion(DataSetVersionStatus dataSetVersionStatus)
+        {
+            ReleaseFile releaseFile = DataFixture.DefaultReleaseFile()
+                .WithReleaseVersion(DataFixture.DefaultReleaseVersion()
+                    .WithPublication(DataFixture
+                        .DefaultPublication()))
+                .WithFile(DataFixture.DefaultFile());
+
+            await AddTestData<ContentDbContext>(context =>
+            {
+                context.ReleaseFiles.Add(releaseFile);
+            });
+
+            DataSet dataSet = DataFixture
+                .DefaultDataSet()
+                .WithStatusDraft()
+                .WithPublicationId(releaseFile.ReleaseVersion.PublicationId);
+
+            await AddTestData<PublicDataDbContext>(context => context.DataSets.Add(dataSet));
+
+            DataSetVersion dataSetVersion = DataFixture
+                .DefaultDataSetVersion(filters: 1, indicators: 1, locations: 1, timePeriods: 2)
+                .WithVersionNumber(1, 0, 0)
+                .WithStatus(dataSetVersionStatus)
+                .WithReleaseFileId(releaseFile.Id)
+                .WithDataSet(dataSet)
+                .WithImports(() => DataFixture
+                    .DefaultDataSetVersionImport()
+                    .Generate(1))
+                .FinishWith(dsv => dsv.DataSet.LatestDraftVersion = dsv);
+
+            await AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.Add(dataSetVersion);
+                context.DataSets.Update(dataSet);
+            });
+
+            releaseFile.File.PublicApiDataSetId = dataSet.Id;
+            releaseFile.File.PublicApiDataSetVersion = dataSetVersion.FullSemanticVersion();
+
+            await AddTestData<ContentDbContext>(context => context.Files.Update(releaseFile.File));
+
+            var dataSetVersionDirectory = _dataSetVersionPathResolver.DirectoryPath(dataSetVersion);
+
+            Directory.CreateDirectory(dataSetVersionDirectory);
+            System.IO.File.WriteAllText(Path.Combine(dataSetVersionDirectory, "version1.txt"), "dummy file text");
```

Comment:
I think you can use `WriteAllTextAsync` here. Alterntively you could create an empty file with

```csharp
await System.IO.File.Create(Path.Combine(dataSetVersionDirectory, "version1.txt")).DisposeAsync();
```

as we aren't particularly bothered about the content.

Reply:
good shout, I've gone for just creating an empty file